### ISSUE TITLE
Lower precision when using hierachical softmax as loss function

### ISSUE THREAD

**Issue by Cli212** [facebookresearch/fastText#1084]
I am using fastText to do text classification and I find the result is extremly different when using hs as loss function comparing with softmax. Keeping other hyper parameters the same, when I used hierachical softmax I got a precision around 0.83, however a 0.91 when using softmax. Can't figure out why.

**Comment by Omerktn** [comment]
Hierachical softmax is faster to evaluate. While softmax is O(n) time, hierarchical softmax is O(logn) time.
But it has a lower precision due to it's speed.
